File: examples/search-algorithms/case-large-found-start.ts
function lcg(seed: number): () => number {
  let state = seed;
  return () => {
    state = (state * 1664525 + 1013904223) & 0xffffffff;
    return (state >>> 0) / 0xffffffff;
  };
}

function shuffledArray(size: number, seed: number): number[] {
  const arr = Array.from({ length: size }, (_, i) => i);
  const rng = lcg(seed);
  for (let i = arr.length - 1; i > 0; i--) {
    const j = Math.floor(rng() * (i + 1));
    [arr[i], arr[j]] = [arr[j], arr[i]];
  }
  return arr;
}

const data = shuffledArray(10000, 123);
const sortedData = [...data].sort((a, b) => a - b);
const target = data[5];

export function createCase() {
  return {
    case: {
      caseId: "large-found-start",
      caseClass: "large",
      name: "Large Array — Target Near Start",
      version: "1.0.0" as const,
      inputs: { data, sortedData, target },
    },
    getInput: async () => ({ data, sortedData, target }),
    getInputs: () => ({ data, sortedData, target }),
  };
}

File: examples/search-algorithms/case_large_found_start.py
"""Large Array — Target Near Start (10000 elements, seed 123, target at index 5)."""

import math


def _lcg(seed):
    state = seed

    def next_val():
        nonlocal state
        state = (state * 1664525 + 1013904223) & 0xFFFFFFFF
        return (state & 0xFFFFFFFF) / 0xFFFFFFFF

    return next_val


def _shuffled_array(size, seed):
    arr = list(range(size))
    rng = _lcg(seed)
    for i in range(len(arr) - 1, 0, -1):
        j = math.floor(rng() * (i + 1))
        arr[i], arr[j] = arr[j], arr[i]
    return arr


_data = _shuffled_array(10000, 123)
_sorted_data = sorted(_data)
_target = _data[5]


def create_case():
    return {
        "case": {
            "caseId": "large-found-start",
            "caseClass": "large",
            "name": "Large Array \u2014 Target Near Start",
            "version": "1.0.0",
            "inputs": {"data": _data, "sortedData": _sorted_data, "target": _target},
        },
        "getInput": lambda: {"data": _data, "sortedData": _sorted_data, "target": _target},
        "getInputs": lambda: {"data": _data, "sortedData": _sorted_data, "target": _target},
    }
